### PR TITLE
pdb: do not raise outcomes.Exit with quit in debug

### PR DESCRIPTION
**Pull request by blueyed**

### CODE REVIEW

File: testing/test_pdb.py [hunk 1]
```diff
@@ -954,3 +954,30 @@ def test_2():
     rest = child.read().decode("utf8")
     assert "no tests ran" in rest
     TestPDB.flush(child)
+
+
+def test_pdb_quit_after_debug(testdir):
```
Comment (nicoddemus) [inline testing/test_pdb.py]: Strange, I get the same behavior as this test shows even in current `features`:

```
λ pytest .tmp\test_quit_db.py
======================================= test session starts =======================================
...
collected 1 item

.tmp\test_quit_db.py
>>>>>>>>>>>>>>>>>>>>>>>>>>>>> PDB set_trace (IO-capturing turned off) >>>>>>>>>>>>>>>>>>>>>>>>>>>>>
--Return--
> c:\pytest\.tmp\test_quit_db.py(6)test_1()->None
-> __import__('pdb').set_trace()
(Pdb) debug foo
ENTERING RECURSIVE DEBUGGER
> <string>(1)<module>()->None
((Pdb)) q
LEAVING RECURSIVE DEBUGGER
(Pdb) q
Exit: Quitting debugger


!!!!!!!!!!!!!!!!!!!!!!!!!!!! _pytest.outcomes.Exit: Quitting debugger !!!!!!!!!!!!!!!!!!!!!!!!!!!!!
================================== no tests ran in 6.57 seconds ===================================
```

What do you get when you execute this test step by step on Linux?

Comment (blueyed) [reply testing/test_pdb.py]: Oh, that's related to pdbpp again - similar to https://github.com/pytest-dev/pytest/pull/4347 I guess.

pdb uses a new Pdb instance inside of `do_debug`, and pdbpp injects its custom `Pdb` into the function, which results in its / pytest's `set_quit` being called then.

Will investigate / adjust.

**Comment by blueyed** [comment]
Adjusted the test.

**Comment by nicoddemus** [comment]
There's some drop in coverage which seems suspect, could you please double check? https://codecov.io/gh/pytest-dev/pytest/compare/b3319a6074636eb651063bff7eec69ec289f1df5...facad4e84d9545bf18a54d2d8d6aa1be6a34a28f/src/src/_pytest/debugging.py

**Comment by blueyed** [comment]
Hmm.. it does not show in the changes tab though (https://codecov.io/gh/pytest-dev/pytest/compare/b3319a6074636eb651063bff7eec69ec289f1df5...facad4e84d9545bf18a54d2d8d6aa1be6a34a28f/changes) - how did you get there?

**Comment by nicoddemus** [comment]
I've clicked in the "codecov/patch" link in the build status, then on `debugging.py` on the heat map on the "Overview" tab.

**Comment by blueyed** [comment]
Coverage appears to be not very reliable for tests involving set_trace, i.e. how could like 172 be covered before, if 171 was not?!  https://codecov.io/gh/pytest-dev/pytest/compare/69a55d334a4c0fe47c469175dc5f077d4f9de6cc...f38c512d99a2e0bd21f3e419217ff06770607966/src/src/_pytest/debugging.py#L165
Might also be more an issue with codecov - would need to investigate into the build logs (where coverage is also dumped with missing lines, but do not really feel like it now ;))

Maybe rebase this after https://github.com/pytest-dev/pytest/pull/4951, where `pdb.set_trace` is mocked for better coverage in some tests already?

**Comment by blueyed** [comment]
Looks like a codecov bug to me, not handling the offset changes in the file properly.

@nicoddemus 
Please merge if you are OK with this.

**Pull request merged**